client/hooks/useSignInWithEmailAndPassword.ts

```diff
@@ -0,0 +1,31 @@
+import { useMutation, UseMutationResult } from '@tanstack/react-query';
+import { signInWithEmailAndPassword, User } from 'firebase/auth';
+
+import { auth } from '@/firebase/firebaseConfig';
+
+interface CreateUserResult {
+  user: User;
+  token: string;
+}
+
+export const useSignInWithEmailAndPasswordMutation = (): UseMutationResult<
+  CreateUserResult,
+  Error,
+  { email: string; password: string }
+> =>
+  useMutation<CreateUserResult, Error, { email: string; password: string }>({
+    mutationFn: async ({ email, password }) => {
+      try {
+        const userCredential = await signInWithEmailAndPassword(
+          auth,
+          email,
+          password,
+        );
+        const user = userCredential.user;
+        const token = await user.getIdToken();
+        return { user, token };
```

Comment:
I wonder the benefit of retrieving firebase tokens from these hooks since we can retrieve it at the axios level. Example see /config -> axios.ts

Reply:
@kinskiwu Agreed. I think we need to generally clean up the hooks/ folder.